src/hooks/useHandleCreateWrapped.tsx

```diff
@@ -678,13 +681,64 @@ async function sui(
   }
 }
 
+async function cosmos(
+  dispatch: any,
+  enqueueSnackbar: any,
+  signedVAA: Uint8Array,
+  foreignAddress: string | null | undefined,
+  sourceChain: ChainId,
+  sourceChainAddress: string
+) {
+  dispatch(setIsCreating(true));
+  let tries = 0;
+  let messageShow = false;
+  const interval = setInterval(async () => {
+    try {
+      if (tries <= 5) {
+        tries++;
+        const txs = await queryWormchain(sourceChainAddress, sourceChain);
+        if (txs.length === 0) {
+          return null;
+        }
+        if (txs.length > 1) {
+          throw new Error("Multiple transactions found");
+        }
+        clearInterval(interval);
+        dispatch(
+          setCreateTx({
+            id: txs[0].hash,
+            block: txs[0].height,
+          })
+        );
+        if (!messageShow) {
+          messageShow = true;
+          enqueueSnackbar(null, {
+            content: <Alert severity="success">Transaction confirmed</Alert>,
+          });
+        }
+      } else {
+        clearInterval(interval);
+        dispatch(setIsCreating(false));
+      }
+    } catch (e) {
+      console.error(e);
+      enqueueSnackbar(null, {
+        content: <Alert severity="error">{parseError(e)}</Alert>,
+      });
+      dispatch(setIsCreating(false));
+    }
+  }, 3500);
```

Comment:
maybe make it exponential could help reducing the amount of calls

Reply:
Done!